File: Backend/UserDataService/src/main/java/com/user/dto/LeaderBoardOnRewardPoints.java
package com.user.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@NoArgsConstructor
@AllArgsConstructor
@Data
public class LeaderBoardOnRewardPoints {
	private String name;
	private String city;
	private Double totalRewardPoints;
	private boolean isCurrentUser = false;
	
	public LeaderBoardOnRewardPoints(String name, String city, Double totalRewardPoints) {
		this.name = name;
		this.city = city;
		this.totalRewardPoints = totalRewardPoints;
	}
}

File: FrontEnd/Self-Tracker/src/app/models/leader-board-on-reward-points.ts
export class LeaderBoardOnRewardPoints {
  name: string;
  city: string;
  totalRewardPoints: number;
  isCurrentUser: boolean;

  constructor(name: string, city: string, totalRewardPoints: number, isCurrentUser: boolean) {
    this.name = name;
    this.city = city;
    this.totalRewardPoints = totalRewardPoints;
    this.isCurrentUser = isCurrentUser;
  }
}
